flow/core/kernel/vehicle/traci.py

```diff
@@ -421,23 +426,51 @@ def get_initial_speed(self, veh_id):
     def get_observed_pedestrians(self, veh_id, pedestrians, radius=50):
         position = self.get_orientation(veh_id)[:2]
         orientation = self.get_orientation(veh_id)[2]
-        orientation += 90 # orientation is 0 when facing North in SUMO, but 0 when facing EAST in util.py
         observed_pedestrians = []
         for ped_id in pedestrians.get_ids():
             if util.observed(position, orientation, pedestrians.get_position(ped_id), looking_distance=radius):
                 observed_pedestrians.append(ped_id)
         return observed_pedestrians
 
-    def get_observed_vehicles(self, veh_id, vehicles, radius=50):
+    def get_observed_vehicles(self, veh_id, radius=50):
+        position = self.get_orientation(veh_id)[:2]
+        orientation = self.get_orientation(veh_id)[2]
```

Comment:
Hmm, it might be worthwhile to step up and have get_orientation to return a dict so we don't have magic numbers here. It's also slightly confusing that get_orientation returns a position